overlord/snapstate/snapmgr.go

```diff
@@ -152,6 +152,9 @@ type SnapState struct {
 	// or otherwise removed from the snap in a future refresh.
 	LastActiveDisabledServices []string `json:"last-active-disabled-services,omitempty"`
 
+	ServicesEnabledByHooks  []string `json:"services-enabled-by-hooks,omitempty"`
```

Comment:
some doc comments mentioning  "tracking services enabled/disabled by hooks" or similar would be good